metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/ScriptParser.java

```diff
@@ -0,0 +1,175 @@
+package org.apache.metron.parsers;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Serializable;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+import javax.script.Invocable;
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineFactory;
+import javax.script.ScriptEngineManager;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.metron.common.Constants;
+import org.apache.metron.parsers.interfaces.MessageParser;
+import org.json.simple.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+
+public class ScriptParser implements MessageParser<JSONObject>,Serializable{
+	
+	protected static final Logger LOG = LoggerFactory.getLogger(ScriptParser.class);
+	protected String scriptPath;
+	protected ScriptEngine engine;
+	protected String parseFunction;
+	protected String language;
+	protected String commonScript="/scripts/";
+	protected List<String> timeFields = new ArrayList<>();
+	protected String timestampField;
+	protected SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.S z");
+
+	@Override
+	public void configure(Map<String, Object> config) {
+		// TODO Auto-generated method stub
+		this.scriptPath=(String) config.get("path");
+		this.parseFunction=(String)config.get("function");
+		this.language=(String)config.get("language");
+		this.commonScript=this.commonScript+language+"/common";
+		if(this.parseFunction==null)
+			this.parseFunction="parse";
+	}
+	//Should this be sent to the interface as a default method?
+	public InputStream openInputStream(String streamName) throws IOException {
+	    FileSystem fs = FileSystem.get(new Configuration());
```

Comment:
So this is following the pattern of what's currently in Grok.  That is being moved to take the grok statements from zookeeper because otherwise you cannot update the parser without bouncing the topology.  Should we follow suit here?